### PR TITLE
Use glob to avoid setting a wildcard route

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/fastify/fastify-static/issues/64.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -94,18 +95,44 @@ function fastifyStatic (fastify, opts, next) {
...
     })
   }
 
+  if (opts.serve !== false) {
+    if (opts.wildcard === undefined || opts.wildcard === true) {
+      fastify.get(prefix + '*', schema, function (req, reply) {
+        pumpSendToReply(req, reply, '/' + req.params['*'])
+      })
+    } else {
+      glob(path.join(sendOptions.root, '**/*'), function (err, files) {
```
Comment (allevo) [inline index.js]: we can use the [sync method](https://www.npmjs.com/package/glob#globsyncpattern-options).

File: README.md [hunk 2]
```diff
@@ -73,6 +73,15 @@ The following options are also supported and will be passed directly to the
...
 
+#### `wildcard`
+
+Default: `true`
+
+If set to `true`, `fastify-static` adds a wildcard route to serve files.
+If set to `false`, `fastify-static` globs the filesystem for all defined
+files in the served folder, and just creates the routes needed for
+those.
+
```
Comment (allevo) [inline README.md]: "The default options of https://www.npmjs.com/package/glob are applied for getting the file list"?

File: index.js (hunk 1 above)
Comment (mcollina) [reply index.js]: why? fastify boots asynchronously, so it should be ok to do this.

Comment (allevo) [reply index.js]: It is simplest.

Comment (mcollina) [reply index.js]: It encourages a wrong pattern of booting synchronously.

File: index.js [hunk 3]
```diff
@@ -94,18 +95,44 @@ function fastifyStatic (fastify, opts, next) {
...
+        if (err) {
+          return next(err)
+        }
+        for (let file of files) {
+          file = file.replace(sendOptions.root, '')
+          const route = (prefix + file).replace(/^\/\//, '/').replace(/\\/, '/')
```
[the remark this answers is not in the document]
Comment (mcollina) [reply index.js]: Yes, removed.

**Comment by allevo** [comment]
There's a bug on windows https://travis-ci.org/fastify/fastify-static/jobs/475673271#L1273

**Comment by cemremengu** [comment]
I will look into it

**Comment by mcollina** [comment]
`wildcard` is not expected to be a string, and we should throw if it's a string. That's the fix I mentioned at the top.. why is this not feasible?

**Pull request merged**